packages/next/src/build/webpack/plugins/define-env-plugin.ts

```diff
@@ -95,6 +95,9 @@ export function getDefineEnv({
       isEdgeServer ? 'edge' : isNodeServer ? 'nodejs' : ''
     ),
     'process.env.NEXT_MINIMAL': JSON.stringify(''),
+    'process.env.__NEXT_WINDOW_HISTORY_SUPPORT': JSON.stringify(
```

Comment:
I'm surprised this is the way we pass information to the client logic but I guess it makes sense

Reply:
This in particular gets replaced to `true` or `false` in the code itself by webpack, which then causes dead code elimination to completely remove the if statement when it's set to `false` 👍